src/global/pid/RichTrack_factory.cc

```diff
@@ -70,14 +70,26 @@ void eicrecon::RichTrack_factory::Process(const std::shared_ptr<const JEvent> &e
     }
   }
 
+  // choose the filter surface: all charged particles that pass through the dRICH vessel will pass
+  // through the backplane of the gas radiator
+  std::shared_ptr<Acts::Surface> filter_surface;
+  for(auto& [output_tag, radiator_tracking_planes] : m_tracking_planes) {
+    if(richgeo::ParseRadiatorName(output_tag, m_log) == richgeo::kGas) {
+      filter_surface = radiator_tracking_planes.back();
```

Comment:
With our current geometry, this change would get _all_ charged particles which pass through either radiator, including the ones that exit out the vessel radial wall (which we may not care about).
```suggestion
      filter_surface = radiator_tracking_planes.front();
```

Reply:
@chchatte92 I leave it up to you to decide which we want.

Reply:
> @chchatte92 I leave it up to you to decide which we want.

Let's stick to `filter_surface=radiator_tracking_planes.back()` . We have intuitive feeling for this scenario.